cmd/solvercli/cmd/balances.go

```diff
@@ -0,0 +1,290 @@
+		if totalUSDValue.Cmp(big.NewFloat(0)) > 0 {
+			fmt.Printf("Total Custom Assets USD Value: %.2f USD\n", totalUSDValue)
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(balancesCmd)
+	balancesCmd.Flags().String("custom-assets", "", "JSON map of chain IDs to denom arrays, e.g. '{\"osmosis\":[\"uosmo\",\"uion\"],\"celestia\":[\"utia\"]}'")
+}
+
+// returns USDC and gas balances for all configured chains
+func getChainBalances(ctx context.Context, evmClientManager evmrpc.EVMRPCClientManager,
+	cctpClientManager *clientmanager.ClientManager) (map[string]*ChainBalance, map[string]*ChainGasBalance, error) {
+	balances := make(map[string]*ChainBalance)
+	gasBalances := make(map[string]*ChainGasBalance)
+	chains := config.GetConfigReader(ctx).Config().Chains
+
+	for _, chain := range chains {
+		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(chain.ChainID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting chain config for %s: %w", chain.ChainID, err)
+		}
+
+		cctpClient, err := cctpClientManager.GetClient(ctx, chain.ChainID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting cctpClient for %s: %w", chain.ChainID, err)
+		}
+
+		var balance *big.Int
+		switch chainConfig.Type {
+		case config.ChainType_EVM:
+			client, err := evmClientManager.GetClient(ctx, chain.ChainID)
+			if err != nil {
+				return nil, nil, fmt.Errorf("getting evm client for chain %s: %w", chain.ChainID, err)
+			}
+			balance, err = client.GetUSDCBalance(ctx, chainConfig.USDCDenom, chainConfig.SolverAddress)
+			if err != nil {
+				return nil, nil, fmt.Errorf("getting balance for %s: %w", chain.ChainID, err)
+			}
+
+		case config.ChainType_COSMOS:
+			balance, err = cctpClient.Balance(ctx, chainConfig.SolverAddress, chainConfig.USDCDenom)
+			if err != nil {
+				return nil, nil, fmt.Errorf("getting balance for %s: %w", chain.ChainID, err)
+			}
+		}
+
+		gasBalance, err := cctpClient.SignerGasTokenBalance(ctx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting gas balance for %s: %w", chain.ChainID, err)
+		}
+
+		warningThreshold, criticalThreshold, err := config.GetConfigReader(ctx).GetGasAlertThresholds(chain.ChainID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting gas alert thresholds for %s: %w", chain.ChainID, err)
+		}
+
+		gasBalances[chain.ChainID] = &ChainGasBalance{
+			ChainID:           chain.ChainID,
+			Balance:           gasBalance,
+			Symbol:            chainConfig.GasTokenSymbol,
+			Decimals:          chainConfig.GasTokenDecimals,
+			WarningThreshold:  warningThreshold,
+			CriticalThreshold: criticalThreshold,
+		}
+
+		balances[chain.ChainID] = &ChainBalance{
+			ChainID:    chain.ChainID,
+			AssetDenom: chainConfig.USDCDenom,
+			Balance:    balance,
+			Symbol:     "USDC",
+			Decimals:   CCTP_TOKEN_DECIMALS,
+		}
+	}
+
+	return balances, gasBalances, nil
+}
+
+func getCustomAssetsBalances(ctx context.Context,
```

Comment:
why do we rely on two methods of fetching asset balances here, it seems like fetching usdc and gas balances can just rely on the balances endpoint?

Reply:
good point, updated